src/lib_windows.cr

```diff
@@ -0,0 +1,253 @@
+@[CallConvention("X86_StdCall")]
+lib LibWindows
+  STD_INPUT_HANDLE  = 0xFFFFFFF6_u32
+  STD_OUTPUT_HANDLE = 0xFFFFFFF5_u32
+  STD_ERROR_HANDLE  = 0xFFFFFFF4_u32
+
+  FILE_TYPE_UNKNOWN = 0x0000
+  FILE_TYPE_DISK    = 0x0001
+  FILE_TYPE_CHAR    = 0x0002
+  FILE_TYPE_PIPE    = 0x0003
+  FILE_TYPE_REMOTE  = 0x8000
+
+  GENERIC_EXECUTE = 0x20000000
+  GENERIC_WRITE   = 0x40000000
+  GENERIC_READ    = 0x80000000
+
+  CREATE_NEW        = 1
+  CREATE_ALWAYS     = 2
+  OPEN_EXISTING     = 3
+  OPEN_ALWAYS       = 4
+  TRUNCATE_EXISTING = 5
+
+  FILE_FLAG_OVERLAPPED = 0x40000000
+
+  # source https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751(v=vs.85).aspx
+  alias Long = Int32
+  alias Word = UInt16
+  alias WChar = UInt16
+  alias DWord = UInt32
+  alias Handle = Void*
+  alias SizeT = UInt64
+  alias BOOL = Int32 # FIXME maybe it need to be removed because it can be confused with Bool
+
+  INVALID_HANDLE_VALUE = Pointer(Void).new((-1).to_u64)
+  INFINITY             = 0xFFFFFFFF_u32
+
+  fun duplicate_handle = DuplicateHandle(source_process : Handle, source : Handle, target_process : Handle, target : Handle*, desired_access : DWord, inherit_handle : Bool, options : DWord) : Bool
+
+  struct Overlapped
+    status : SizeT
+    bytes_transfered : SizeT
+    offset : UInt64
+    event : Handle
+  end
+
+  struct SecurityAttributes
+    length : DWord
+    security_descriptors : Void*
+    inherit_handle : BOOL
+  end
+
+  fun get_std_handle = GetStdHandle(std_handle : DWord) : Handle
+  fun get_file_type = GetFileType(file : Handle) : DWord
+  fun create_file = CreateFileA(filename : UInt8*, access : DWord, sharemode : DWord, security_attributes : SecurityAttributes*, creation : DWord, flags : DWord, template : Handle) : Handle
+  fun read_file = ReadFile(file : Handle, buffer : UInt8*, size : DWord, read : DWord*, overlapped : Overlapped*) : Bool
+  fun write_file = WriteFile(file : Handle, buffer : UInt8*, size : DWord, written : DWord*, overlapped : Overlapped*) : Bool
+  fun close_handle = CloseHandle(file : Handle) : Bool
+
+  fun get_current_directory = GetCurrentDirectoryA(size : DWord, buffer : UInt8*) : DWord
+  fun set_current_directory = SetCurrentDirectoryA(path : UInt8*) : BOOL
+  fun create_directory = CreateDirectoryA(path : UInt8*, security_attribute : Void*) : BOOL
+  fun remove_directory = RemoveDirectoryA(path : UInt8*) : BOOL
+  fun get_temp_path = GetTempPathA(len : DWord, buffer : UInt8*) : DWord
+
+  MAX_PATH = 260
+
+  struct FILETIME
+    dwLowDateTime : DWord
+    dwHighDateTime : DWord
+  end
+
+  struct WIN32_FIND_DATA_A
+    dwFileAttributes : DWord
```

Comment:
just my 2 cents. I partly agree. Struct names have to be the same due to Microsoft documentation. But struct member naming can be changed because:
- otherwise they look alien
- it's still easy to find them in documentation due to struct naming
- I don't think this binding will be regenerated eventually (because API aren't changed for decades and other reasons). But even so, it would be easy to keep previous changed names during regeneration.

And more importantly, what to do with the current function naming, for example:
fun duplicate_handle = DuplicateHandle(...)
Shoud I rename duplicate_handle as DuplicateHanle?
Sorry for long reading.